OracleConnectorIT#shouldFailWhenConverterThrowsExceptionForValue fails randomly

Fails randomly with
```
11:38:23,224 [ERROR] io.debezium.connector.oracle.OracleConnectorIT.shouldFailWhenConverterThrowsExceptionForValue -- Time elapsed: 11.19 s <<< ERROR!
java.lang.IllegalStateException: Engine is already being shutting down.
	at io.debezium.embedded.async.AsyncEmbeddedEngine.close(AsyncEmbeddedEngine.java:264)
	at io.debezium.embedded.async.TestingAsyncEmbeddedEngine.close(TestingAsyncEmbeddedEngine.java:34)
	at io.debezium.embedded.AbstractConnectorTest.stopConnector(AbstractConnectorTest.java:162)
	at io.debezium.embedded.AbstractConnectorTest.stopConnector(AbstractConnectorTest.java:147)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

because exception (intentionally) thrown in the test starts engine shutdown and and engine shutdown is also run after each test. If those two overlap, it can result into this exception. We should wait in the test until engine shutdown is finished.

--- a/debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java
+++ b/debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java
@@ -6302,6 +6302,7 @@ public class OracleConnectorIT extends AbstractAsyncEngineConnectorTest {
                                     TestHelper.getDatabaseName())));
 
             assertNoRecordsToConsume();
+            waitForEngineShutdown();
         }
         finally {
             TestHelper.dropTable(connection, "dbz1508");

